pylint/checkers/typecheck.py

```diff
@@ -1013,6 +1016,24 @@ def visit_attribute(self, node: nodes.Attribute) -> None:
 
         function/method, super call and metaclasses are ignored
         """
+
```

Comment:
I think it would be a good thing to isolate the cache updating in a specifically named function si it's explicit that this part is about caching.